kubernetes/cache/cache.go

```diff
@@ -145,7 +147,7 @@ func NewKialiCache(clientFactory kubernetes.ClientFactory, cfg config.Config) (K
 	}
 
 	for cluster, client := range clientFactory.GetSAClients() {
-		cache, err := NewKubeCache(client, cfg)
+		cache, err := NewKubeCache(client, cfg, kialiCacheImpl.deleteNamespace)
```

Comment:
Can we pass `nil` when it's a cluster scoped cache? I really don't want to call this error handler or have any side effects when there's a cluster scoped cache.

Reply:
We know we are cluster scoped at this point in the code right? I guess "cfg.Deployment.ClusterWideAccess" ??

Reply:
This change is coming when I push my next commit:
```diff
diff --git a/kubernetes/cache/cache.go b/kubernetes/cache/cache.go
index eb5edf1d7..e306546de 100644
--- a/kubernetes/cache/cache.go
+++ b/kubernetes/cache/cache.go
@@ -147,7 +147,12 @@ func NewKialiCache(clientFactory kubernetes.ClientFactory, cfg config.Config) (K
        }
 
        for cluster, client := range clientFactory.GetSAClients() {
-               cache, err := NewKubeCache(client, cfg, kialiCacheImpl.deleteNamespace)
+               // we only need our deleteNamespace function called when an error occurs in a namespace-scoped cache
+               var errHandler ErrorHandler
+               if !cfg.Deployment.ClusterWideAccess {
+                       errHandler = kialiCacheImpl.deleteNamespace
+               }
+               cache, err := NewKubeCache(client, cfg, errHandler)
```

Reply:
done.